Nodejs/Product/Nodejs/TestFrameworks/mocha/mocha.js

```diff
@@ -53,33 +57,68 @@ var run_tests = function (testName, testFile, workingFolder, projectFolder) {
         return;
     }
 
-    var mocha = new Mocha();
-    mocha.ui('tdd');
-    //set timeout to 10 minutes, because the default of 2 sec might be too short (TODO: make it configurable)
-    mocha.suite.timeout(600000);
+    var mocha = initializeMocha(Mocha, projectFolder);
+
     if (testName) {
         mocha.grep(testName);
     }
     mocha.addFile(testFile);
 
-    // Choose 'tap' rather than 'min' or 'xunit'. The reason is that
-    // 'min' produces undisplayable text to stdout and stderr under piped/redirect, 
-    // and 'xunit' does not print the stack trace from the test.
-    mocha.reporter('tap');
-
     mocha.run(function (code) {
         process.exit(code);
     });
 };
 
+function logError(errorMessage) {
+    console.log("NTVS_ERROR: " + errorMessage);
+}
+
 function detectMocha(projectFolder) {
     try {
         var Mocha = new require(projectFolder + '\\node_modules\\mocha');
         return Mocha;
     } catch (ex) {
-        console.log("NTVS_ERROR:Failed to find Mocha package.  Mocha must be installed in the project locally.  Mocha can be installed locally with the npm manager via solution explorer or with \".npm install mocha\" via the Node.js interactive window.");
+        logError("Failed to find Mocha package.  Mocha must be installed in the project locally.  Mocha can be installed locally with the npm manager via solution explorer or with \".npm install mocha\" via the Node.js interactive window.");
         return null;
     }
 }
 
+function initializeMocha(Mocha, projectFolder) {
+    var mocha = new Mocha();
+    applyMochaOptions(mocha, getMochaOptions(projectFolder));
+    return mocha;
+}
+
+function applyMochaOptions(mocha, options) {
+    if (options) {
+        for (var opt in options) {
+            var mochaOpt = mocha[opt];
+            var optValue = options[opt];
+
+            if (typeof mochaOpt == 'function') {
+                try {
+                    mochaOpt.call(mocha, optValue);
+                } catch (e) {
+                    logError("Could not set mocha option '" + opt + "' with value '" + optValue + "' due to error: " + e);
+                }
+            }
+        }
+    }
+}
+
+function getMochaOptions(projectFolder) {
+    var mochaOptions = defaultMochaOptions;
+    try {
+        var options = require(projectFolder + '\\test\\mocha.json');
+        options = options || {};
+        for (var opt in options) {
+            mochaOptions[opt] = options[opt];
+        }
+    } catch (ex) {
+        console.log("NTVS: mocha.json options file not found. Using default values.");
```

Comment:
This should be console.error("NTVS_ERROR:", blah), otherwise the test reporter won't pick it up (it's a bit confusing because you see it in debug mode.)